src/main/java/org/candlepin/resource/CloudRegistrationResource.java

```diff
@@ -160,6 +163,52 @@ else if (version == 2) {
         }
     }
 
+    @Override
+    public List<String> cancelCloudAccountJobs(String cloudAccountId) {
+
+        if (cloudAccountId == null || cloudAccountId.isBlank()) {
+            throw new BadRequestException(this.i18n.tr("Cloud account ID is null or empty"));
+        }
+
+        AsyncJobStatusQueryArguments asyncJobStatusQueryArguments = new AsyncJobStatusQueryArguments();
+        asyncJobStatusQueryArguments.setJobKeys(CloudAccountOrgSetupJob.JOB_KEY)
+            .setJobStates(AsyncJobStatus.JobState.CREATED, AsyncJobStatus.JobState.QUEUED,
+                AsyncJobStatus.JobState.RUNNING, AsyncJobStatus.JobState.SCHEDULED,
+                AsyncJobStatus.JobState.WAITING, AsyncJobStatus.JobState.FAILED_WITH_RETRY);
+        List<AsyncJobStatus> jobs = this.jobManager.findJobs(asyncJobStatusQueryArguments);
+
+        jobs = jobs.stream()
+            .filter(job -> job.getJobArguments().getAsString("cloud_account_id").equals(cloudAccountId))
+            .collect(Collectors.toList());
```

Comment:
Depending on when this is run, the query could fetch a very large amount of job statuses into memory.
I think using this one is preferred, because it will only fetch the jobs for that cloud account id:
```
Map<String,String> jobArgs = Map.of("cloud_account_id", cloudAccountId);
AsyncJobStatusCurator.fetchJobIdsByArguments(CloudAccountOrgSetupJob.JOB_KEY, jobArgs);
```
and then you can filter based on the job non-terminal job states

Reply:
done